File: benchmarks/prime.go
package main

import (
	"fmt"
	"time"
)

func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	
	for i := 2; i*i <= n; i++ {
		if n%i == 0 {
			return false
		}
	}
	
	return true
}

func countPrimes(limit int) int {
	count := 0
	for i := 2; i < limit; i++ {
		if isPrime(i) {
			count++
		}
	}
	return count
}

func main() {
	start := time.Now()
	result := countPrimes(10000)
	duration := time.Since(start)
	
	fmt.Printf("Go Primes up to 10000: %d\n", result)
	fmt.Printf("Duration: %v\n", duration)
}

File: benchmarks/prime.py
#!/usr/bin/env python3
import time

def is_prime(n):
    if n < 2:
        return False
    
    i = 2
    while i * i <= n:
        if n % i == 0:
            return False
        i += 1
    
    return True

def count_primes(limit):
    count = 0
    for i in range(2, limit):
        if is_prime(i):
            count += 1
    return count

def main():
    start = time.time()
    result = count_primes(10000)
    end = time.time()
    duration = (end - start) * 1000  # Convert to milliseconds
    
    print(f"Python Primes up to 10000: {result}")
    print(f"Duration: {duration:.2f} ms")

if __name__ == "__main__":
    main()
